runtime/vm/jvminit.c

```diff
@@ -8034,10 +8034,12 @@ setThreadNameAsyncHandler(J9VMThread *currentThread, IDATA handlerKey, void *use
 
 	if (shouldUpdateName) {
 		/* don't allow another thread to change the name while I am reading it */
-		threadLock = (j9object_t) objectMonitorEnter(currentThread, threadLock);
-		if (threadLock == NULL) {
-			/* We may be out of memory - try again later */
-			J9SignalAsyncEvent(javaVM, currentThread, handlerKey);
+		threadLock = (j9object_t)objectMonitorEnter(currentThread, threadLock);
```

Comment:
likewise, this only occurs on thread startup so it wont be possible in single thread mode

Reply:
Removed for `setThreadNameAsyncHandler()`.